src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -546,61 +546,113 @@ internal async void ExecutePackageDownload(string name, PackageVersion package,
                 var immediateUninstalls = new List<Package>();
                 var builtinPackages = new List<Package>();
 
-                // if a package is already installed we need to uninstall it, allowing
-                // the user to cancel if they do not want to uninstall the package
-                var duplicateLocalPackages = dependencyVersionHeaders.Select(dep => localPkgs.FirstOrDefault(v => v.Name == dep.name));
-                foreach (var localPkg in duplicateLocalPackages)
-                {
-                    if (localPkg == null) continue;
+                // if the new package has one or more dependencies that are already installed
```

Comment:
this method seems really long already, maybe we could split out this loop into a small function with a comment that explains how it modifies or does not modify the `dependencyVersionHeaders`